include/amqpcpp/address.h

```diff
@@ -113,8 +123,32 @@ class Address
         // find out where the vhost is set (starts with a slash)
         const char *slash = (const char *)memchr(data, '/', last - data);
 
+        // if there is a slash, we search for the ? for extra options
```

Comment:
Why only if there is a slash?
- amqp://localhost/vhost?param=123
- amqp://localhost/?param=123
- amqp://localhost?param=123

Are all of the above cases covered?